sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -0,0 +1,428 @@
+                break;
+        }
+    }
+
+    /**
+     * Helper method to verify TextBatchStatistics.
+     *
+     * @param expectedStatistics
+     * @param actualStatistics
+     */
+    private static void validateBatchStatistics(TextBatchStatistics expectedStatistics,
+                                                TextBatchStatistics actualStatistics) {
+        assertEquals(expectedStatistics.getDocumentCount(), actualStatistics.getDocumentCount());
+        assertEquals(expectedStatistics.getErroneousDocumentCount(), actualStatistics.getErroneousDocumentCount());
+        assertEquals(expectedStatistics.getValidDocumentCount(), actualStatistics.getValidDocumentCount());
+        assertEquals(expectedStatistics.getTransactionCount(), actualStatistics.getTransactionCount());
+    }
+
+    /**
+     * Helper method to verify the error document.
+     *
+     * @param expectedError the Error returned from the service.
+     * @param actualError the Error returned from the API.
+     */
+    static void validateErrorDocument(Error expectedError, Error actualError) {
+        assertEquals(expectedError.getCode(), actualError.getCode());
+        assertEquals(expectedError.getMessage(), actualError.getMessage());
+        assertEquals(expectedError.getTarget(), actualError.getTarget());
+        assertEquals(expectedError.getInnererror(), actualError.getInnererror());
+    }
+
+    /**
+     * Helper method to verify TextDocumentStatistics.
+     *
+     * @param expected the expected value for TextDocumentStatistics.
+     * @param actual the value returned by API.
+     */
+    private static void validateDocumentStatistics(TextDocumentStatistics expected, TextDocumentStatistics actual) {
+        assertEquals(expected.getCharacterCount(), actual.getCharacterCount());
+        assertEquals(expected.getTransactionCount(), actual.getTransactionCount());
+    }
+
+    /**
+     * Helper method to validate a single detected language.
+     *
+     * @param expectedLanguage detectedLanguage returned by the service.
+     * @param actualLanguage detectedLanguage returned by the API.
+     */
+    static void validatePrimaryLanguage(DetectedLanguage expectedLanguage, DetectedLanguage actualLanguage) {
+        assertEquals(expectedLanguage.getIso6391Name(), actualLanguage.getIso6391Name());
+        assertEquals(expectedLanguage.getName(), actualLanguage.getName());
+        assertEquals(expectedLanguage.getScore(), actualLanguage.getScore());
+    }
+
+    /**
+     * Helper method to validate the list of detected languages.
+     *
+     * @param expectedLanguageList detectedLanguages returned by the service.
+     * @param actualLanguageList detectedLanguages returned by the API.
+     */
+    static void validateDetectedLanguages(List<DetectedLanguage> expectedLanguageList,
+                                          List<DetectedLanguage> actualLanguageList) {
+        for (int i = 0; i < expectedLanguageList.size(); i++) {
```

Comment:
This assumes that the order of your expectedLanguage is the same as actualLanguageList. What if they returned "es", "en" rather than "en", "es"? Your test would fail.

Reply:
sorted both lists on basis of name and then comparing.